### PR TITLE
[DS-473] pipeline WIOR

### PR DESCRIPTION
**Pull request by cvriel**

### CODE REVIEW

File: src/dags/wior.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+dag_id = "wior"
+variables = Variable.get(dag_id, deserialize_json=True)
+data_endpoint = variables["data_endpoints"]["wfs"]
+tmp_dir = f"{SHARED_DIR}/{dag_id}"
+data_file = f"{tmp_dir}/{dag_id}.geojson"
+env = Env()
+db_conn = DatabaseEngine()
+password = env("AIRFLOW_CONN_WIOR_PASSWD")
+user = env("AIRFLOW_CONN_WIOR_USER")
+base_url = env("AIRFLOW_CONN_WIOR_BASE_URL")
```
Comment (gkoller) [inline src/dags/wior.py:53]: Suggestion:

```python
from more_ds.network.url import URL
base_url = URL(env("AIRFLOW_CONN_WIOR_BASE_URL"))
```

File: src/dags/wior.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+class DataSourceError(Exception):
+    pass
+
+
+# data connection
+def get_data() -> None:
+    """calling the data endpoint"""
+    # get data
+    data_url = f"{base_url}/{data_endpoint}"
```
Comment (gkoller) [inline src/dags/wior.py:76]: If you using the `URL` class (see suggestion: https://github.com/Amsterdam/dataservices-airflow/pull/197/files#r580328716) Then this becomes:
```python
data_url = base_url / data_endpoint
```

File: src/dags/wior.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    data_request = requests.get(data_url, auth=HTTPBasicAuth(user, password))
+    # store data
+    if data_request.status_code == 200:
+        try:
+            data = data_request.json()
+        except json.decoder.JSONDecodeError:
+            logger.error(f"Failed to convert request output to json for url {data_url}")
```
Comment (gkoller) [inline src/dags/wior.py:83]: Use [logger.exception](https://docs.python.org/3.8/library/logging.html?highlight=logger%20exception#logging.Logger.exception) to have exception info added to the logging message automatically. If that's not what you want, stick to `error`.

File: src/dags/wior.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+total_checks = []
+count_checks = []
+geo_checks = []
+
+
+# define logger for output to console
+logger = logging.getLogger(__name__)
+
+
+# needed to put quotes on elements in geotypes for SQL_CHECK_GEO
```
Comment (jjmurre) [inline src/dags/wior.py:63]: This function is appearing in a lot of DAGs. Better to put it in a central place, eg. the `common` module.

File: src/dags/wior.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    slack_at_start = MessageOperator(
+        task_id="slack_at_start",
+        http_conn_id="slack",
+        webhook_token=slack_webhook_token,
+        message=f"Starting {dag_id} ({DATAPUNT_ENVIRONMENT})",
+        username="admin",
+    )
+
+    # 2. Create temp directory to store files
+    mkdir = BashOperator(task_id="mkdir", bash_command=f"mkdir -p {tmp_dir}")
```
Comment (jjmurre) [inline src/dags/wior.py:109]: @ngaranko Has created an ENV var vor the temporary dir. (No sure if that is merged already). Better to use that ENV var.

File: src/dags/wior.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    # get data
+    data_url = f"{base_url}/{data_endpoint}"
+    data_request = requests.get(data_url, auth=HTTPBasicAuth(user, password))
+    # store data
+    if data_request.status_code == 200:
+        try:
+            data = data_request.json()
+        except json.decoder.JSONDecodeError:
+            logger.error(f"Failed to convert request output to json for url {data_url}")
+            raise
```
Comment (gkoller) [inline src/dags/wior.py:84]: Generally it is a good idea to use exception chaining;

```python
except json.decoder.JSONDecodeError as jde:
    ...
    raise from jde
```

**Review by gkoller — APPROVED** [review]

**Review by jjmurre — APPROVED** [review]
Few nitpicks. For the rest, LGTM!

File: src/dags/wior.py (hunk 5 above)
Comment (ngaranko) [reply src/dags/wior.py:109]: It is used here: https://github.com/Amsterdam/dataservices-airflow/pull/197/files#diff-bdad4f2581886fbd705c54ceba3027c7c835a8be667ec71c9951d131e071d50cR47
:heavy_check_mark:

File: src/dags/wior.py (hunk 1 above)
Comment (cvriel) [reply src/dags/wior.py]: Ja natuurlijk, more_ds is erg handig, aangepast

File: src/dags/wior.py (hunk 2 above)
Comment (cvriel) [reply src/dags/wior.py]: Aangepast

File: src/dags/wior.py (hunk 3 above)
Comment (cvriel) [reply src/dags/wior.py]: Ik heb exception gebruikt, past beter

File: src/dags/wior.py (hunk 6 above)
Comment (cvriel) [reply src/dags/wior.py]: Explicieter :+1:  aangepast

File: src/dags/wior.py (hunk 4 above)
Comment (cvriel) [reply src/dags/wior.py:60]: Goeie, toevoegd en nu uit `common` gehaald

File: src/dags/wior.py (hunk 5 above)
Comment (cvriel) [reply src/dags/wior.py:107]: Check :+1:

**Pull request merged**